github-connector/internal/handlers/webhookhandler_test.go

```diff
@@ -0,0 +1,145 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/go-github/github"
+	"github.com/kyma-incubator/hack-showcase/github-connector/internal/handlers/mocks"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type toJSON struct {
+	TestJSON string `json:TestJSON"`
+}
+
+//createRequest creates an HTTP request for test purposes
+func createRequest(t *testing.T) *http.Request {
+
+	payload := toJSON{TestJSON: "test"}
+	toSend, err := json.Marshal(payload)
+	require.NoError(t, err)
+
+	req := httptest.NewRequest(http.MethodPost, "/webhook", bytes.NewBuffer(toSend))
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Hub-Signature", "test")
+
+	return req
+}
+
+func TestWebhookHandler_TestBadSecret(t *testing.T) {
+	t.Run("should respond with 401 status code", func(t *testing.T) {
+		// given
+
+		payload := toJSON{TestJSON: "test"}
+		toSend, err := json.Marshal(payload)
+		require.NoError(t, err)
+
+		req := httptest.NewRequest(http.MethodPost, "/webhook", bytes.NewBuffer(toSend))
+		req.Header.Set("Content-Type", "application/json")
+
+		rr := httptest.NewRecorder()
+		mockHandler := &mocks.Validator{}
+
+		mockHandler.On("GetToken").Return("test")
+		mockHandler.On("ValidatePayload", req, []byte("test")).Return(nil, errors.New("failed"))
+
+		// when
+		wh := NewWebHookHandler(mockHandler)
+
+		handler := http.HandlerFunc(wh.HandleWebhook)
+		handler.ServeHTTP(rr, req)
+		// then
```

Comment:
please unify spacing in this file (for example: one empty line before //given //when //then statements)